pkg/kubelet/rocket/rocket.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	if len(output) == 0 { // No containers running.
+		return nil, nil
+	}
+	for _, line := range strings.Split(strings.TrimSpace(string(output)), "\n") {
+		tuples := strings.Split(strings.TrimSpace(line), "\t")
+		if len(tuples) != 3 { // HARDCODE, non state line.
+			continue
+		}
+		status[tuples[0]] = &PodStatus{
+			State: tuples[2],
+		}
+	}
+	return status, nil
+}
+
+func (r *RocketRuntime) unitFileToPod(fname string) (*api.Pod, error) {
+	f, err := os.Open(fname)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var pod api.Pod
+	opts, err := unit.Deserialize(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var rktID string
+	for _, opt := range opts {
+		if opt.Section != "X-K8S" {
+			continue
+		}
+		if opt.Name == "RocketID" {
+			rktID = opt.Value
+		}
+		if opt.Name == "POD" {
+			// NOTE: In fact we unmarshal from a serialized
+			// api.BoundPod type here.
+			err = json.Unmarshal([]byte(opt.Value), &pod)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// TODO(yifan): Cache the list result
+	podState := "unknown"
+	pods, err := r.getPodsStatus()
+	if err == nil {
+		if _, found := pods[rktID]; !found {
+			return nil, fmt.Errorf("pod not running from rkt list")
+		}
+		podState = pods[rktID].State
+	}
+
+	fmt.Println("pod state", rktID, podState)
+
+	pod.Status.Info = make(map[string]api.ContainerStatus)
+	for _, container := range pod.Spec.Containers {
+		switch podState {
+		case "running":
+			pod.Status.Info[container.Name] = api.ContainerStatus{
+				State: api.ContainerState{
+					Running: &api.ContainerStateRunning{},
+				},
+			}
+		case "embryo", "preparing", "prepared":
+			pod.Status.Info[container.Name] = api.ContainerStatus{
+				State: api.ContainerState{
+					Waiting: &api.ContainerStateWaiting{},
+				},
+			}
+		case "exited", "deleting", "gone":
+			pod.Status.Info[container.Name] = api.ContainerStatus{
+				State: api.ContainerState{
+					Termination: &api.ContainerStateTerminated{},
+				},
+			}
+		default:
+			panic("Unexpected state")
+		}
+	}
+	return &pod, nil
+}
+
+func (r *RocketRuntime) preparePod(pod *api.BoundPod) (string, error) {
+	units, err := r.podToUnits(pod)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the pod's uuid.
+	var images []string
+	for _, c := range pod.Spec.Containers {
+		images = append(images, c.Image)
+	}
+	output, err := r.runCommand(true, "prepare", images...)
+	if err != nil {
+		return "", err
+	}
+	uuid := strings.TrimSpace(string(output))
+	if uuid == "" {
+		panic("expect uuid returned, but get nothing")
+	}
+	fmt.Println("Prepare uuid:", uuid)
+
+	units = append(units,
+		&unit.UnitOption{
+			Section: "X-K8S",
+			Name:    "RocketID",
+			Value:   uuid,
+		},
+		&unit.UnitOption{
+			Section: "Service",
+			Name:    "ExecStart",
+			Value:   r.constructRunCmd(uuid),
+		},
+	)
+
+	// Save the unit file.
+	name := fmt.Sprintf("K8S_%s_%s.service", pod.Name, pod.Namespace)
+	unitFile, err := os.Create(path.Join(tmpDirPath, name))
```

Comment:
Use the builtin http://golang.org/pkg/io/ioutil/#TempFile for management of temporary files like this.